client/src/components/hooks/useGetWallets.js

```diff
@@ -1,42 +1,35 @@
 import { useState, useCallback } from 'react';
-import { WalletDescriptor } from 'casper-storage';
-import isNil from 'lodash-es/isNil';
-import get from 'lodash-es/get';
-import { formatAccountName } from '@cd/helpers/format';
+import findLastIndex from 'lodash-es/findLastIndex';
+import { loadBalanceWallets } from '@cd/helpers/wallet';
 
-import { getUserHDWallets, addWalletAccount } from '@cd/hooks/useServiceWorker';
-import { getAccounts } from '@cd/services/userServices';
-import { convertBalanceFromHex } from '@cd/helpers/balance';
+import { getUserHDWallets, generateWallets, removeWalletsByPaths } from '@cd/hooks/useServiceWorker';
 
 const useGetWallets = () => {
 	const [isLoading, setIsLoading] = useState(false);
 	const [wallets, setWallets] = useState([]);
 
+	const generateBalanceWallets = useCallback(async () => {
+		const wallets = await generateWallets(10);
+		const balanceWallets = await loadBalanceWallets(wallets);
+		const foundLastIndex = findLastIndex(balanceWallets, wallet => {
+			return parseInt(wallet.balance, 10) > 0;
+		});
+
+		const ids = balanceWallets.slice(foundLastIndex + 1).map(wallet => wallet._id);
+		if (ids.length > 0) {
+			await removeWalletsByPaths(ids);
```

Comment:
Not really declarative at this line, because we're passing IDs and call it `paths` ?

Reply:
In general we should consider moving this kind of logic into Controller, this makes it easier for UT.